### PR TITLE
mpich: add pmi=cray support

### PR DESCRIPTION
**Pull request by haampie**
After marking `cray-pmi` as external:

```yaml
packages:
  cray-pmi:
    externals:
    - spec: cray-pmi@5.0.17
      prefix: /opt/cray/pe/pmi/5.0.17
```

You can now install

```
spack install something ^mpich pmi=cray
```

and

```console
srun $(spack location -i something)/bin/your_app
```

### CODE REVIEW

**Comment by alalazo** [comment]
@spackbot maintainers

**Review by alalazo — APPROVED** [review]
Waiting to check if @lukebroskop or `mpich` maintainers want to comment, but otherwise LGTM

**Review by raffenet — APPROVED** [review]

**Comment by lukebroskop** [comment]
@alalazo, LGTM.

**Pull request merged**
